src/pybind/mgr/cephadm/module.py

```diff
@@ -2594,6 +2632,43 @@ def _create_rbd_mirror(self, daemon_id, host):
     def apply_rbd_mirror(self, spec):
         return self._apply(spec)
 
+    def _generate_nfs_config(self, daemon_type, daemon_id, host):
+        # type: (str, str, str) -> Tuple[Dict[str, Any], List[str]]
+        deps = [] # type: List[str]
+
+        # find the matching NFSServiceSpec
+        # TODO: find the spec and pass via _create_daemon instead ??
+        service_name = self.get_service_name(daemon_type, daemon_id, host)
+        specs = self.spec_store.find(service_name)
```

Comment:
this part feels fragile, but I don't want todo a bunch of rework for the existing params passed to `_create_daemon` et al.   ...